dagger/run_dbt_spark_tests.py

```diff
@@ -0,0 +1,139 @@
+import os
+
+import argparse
+import sys
+
+import anyio as anyio
+import dagger as dagger
+from dotenv import find_dotenv, load_dotenv
+
+PG_PORT = 5432
+load_dotenv(find_dotenv("test.env"))
+DEFAULT_ENV_VARS = {
+    "DBT_TEST_USER_1": os.getenv("DBT_TEST_USER_1", "buildbot+dbt_test_user_1@dbtlabs.com"),
+    "DBT_TEST_USER_2": os.getenv("DBT_TEST_USER_2", "buildbot+dbt_test_user_2@dbtlabs.com"),
+    "DBT_TEST_USER_3": os.getenv("DBT_TEST_USER_3", "buildbot+dbt_test_user_3@dbtlabs.com"),
+}
+
+
+def env_variables(envs: dict[str, str]):
+    def env_variables_inner(ctr: dagger.Container):
+        for key, value in envs.items():
+            ctr = ctr.with_env_variable(key, value)
+        return ctr
+
+    return env_variables_inner
+
+
+def get_databricks_env_vars():
+    return {
+        "DBT_DATABRICKS_TOKEN": os.environ["DBT_DATABRICKS_TOKEN"],
+        "DBT_DATABRICKS_HOST_NAME": os.environ["DBT_DATABRICKS_HOST_NAME"],
+        "DBT_DATABRICKS_ENDPOINT": os.environ["DBT_DATABRICKS_ENDPOINT"],
+        "DBT_DATABRICKS_CLUSTER_NAME": os.environ["DBT_DATABRICKS_CLUSTER_NAME"],
+        "ODBC_DRIVER": "/opt/simba/spark/lib/64/libsparkodbc_sb64.so",
+    }
+
+
+async def get_postgres_container(client: dagger.Client) -> (dagger.Container, str):
+    ctr = await (
+        client.container()
+        .from_("postgres:13")
+        .with_env_variable("POSTGRES_PASSWORD", "postgres")
+        .with_exposed_port(PG_PORT)
+    )
+
+    return ctr, "postgres_db"
+
+
+async def get_spark_container(client: dagger.Client) -> (dagger.Container, str):
+    spark_dir = client.host().directory("./dagger/spark-container")
+    spark_ctr = (
+        client.container()
+        .from_("eclipse-temurin:8-jre")
+        .with_directory("/spark_setup", spark_dir)
+        .with_env_variable("SPARK_HOME", "/usr/spark")
+        .with_env_variable("PATH", "/usr/spark/bin:/usr/spark/sbin:$PATH", expand=True)
+        .with_file(
+            "/scripts/entrypoint.sh",
+            client.host().file("./dagger/spark-container/entrypoint.sh"),
+            permissions=755,
+        )
+        .with_file(
+            "/scripts/install_spark.sh",
+            client.host().file("./dagger/spark-container/install_spark.sh"),
+            permissions=755,
+        )
+        .with_exec(["./spark_setup/install_spark.sh"])
+        .with_file("/usr/spark/conf/hive-site.xml", spark_dir.file("/hive-site.xml"))
+        .with_file("/usr/spark/conf/spark-defaults.conf", spark_dir.file("spark-defaults.conf"))
+    )
+
+    # postgres is the metastore here
+    pg_ctr, pg_host = await get_postgres_container(client)
+
+    spark_ctr = (
```

Comment:
Could we name this differently than on line 51?  Took a couple read throughs to realize you're not just overwriting it completely.